src/service/ratelimit.go

```diff
@@ -118,9 +135,16 @@ func (this *service) shouldRateLimitWorker(
 	response.Statuses = make([]*pb.RateLimitResponse_DescriptorStatus, len(request.Descriptors))
 	finalCode := pb.RateLimitResponse_OK
 	for i, descriptorStatus := range responseDescriptorStatuses {
-		response.Statuses[i] = descriptorStatus
-		if descriptorStatus.Code == pb.RateLimitResponse_OVER_LIMIT {
-			finalCode = descriptorStatus.Code
+		if isUnlimited[i] {
+			response.Statuses[i] = &pb.RateLimitResponse_DescriptorStatus{
+				Code:           pb.RateLimitResponse_OK,
+				LimitRemaining: math.MaxUint32,
```

Comment:
Can you document this somewhere? It should probably go directly in the proto docs as a comment? Or potentially also somewhere in the README?

Reply:
Not sure what you mean by the proto docs, I added a comment to the README

Reply:
I mean doing a PR here also: https://github.com/envoyproxy/envoy/blob/7c1991ffe382bacf2bf415a643c5ea7428d0880e/api/envoy/service/ratelimit/v3/rls.proto#L130-L131

This is the canonical service definition for the ratelimit service.